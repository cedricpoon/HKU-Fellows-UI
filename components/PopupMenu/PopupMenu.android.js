import React, { Component } from 'react';
import PropTypes from "prop-types";
import { View, Dimensions, TouchableWithoutFeedback } from 'react-native';
import * as Animatable from 'react-native-animatable';

import styles from './Styles';

class PopupMenu extends Component {
  constructor(props) {
    super(props);
    this.state = { toggleFlag: false };
  }

  toggle = () => {
    const { height } = Dimensions.get('window');
    const { toggleFlag } = this.state;

    const flipFlag = () => { this.setState({ toggleFlag: !this.state.toggleFlag }); };

    if (toggleFlag) {
      if (position.y > height / 2)
        this._menu.fadeOutDown(100).then(flipFlag);
      else
        this._menu.fadeOutUp(100).then(flipFlag);
    } else {
      flipFlag();
    }
  }

  _renderOutsider = () => {
    return (
      <TouchableWithoutFeedback onPress={this.toggle}>
        <View style={styles.outsider} />
      </TouchableWithoutFeedback>
    );
  }

  render() {
    const { children, position } = this.props;
    const { toggleFlag } = this.state;
    const { height } = Dimensions.get('window');

    if (toggleFlag) {
      return(
        <View>
          <Animatable.View
            style={styles.menuAndroid}
            animation={position.y > height / 2 ? "fadeInUp" : "fadeInDown"}
            duration={100}
            ref={ref => { this._menu = ref }}
          >
            { children }
          </Animatable.View>
          {this._renderOutsider()}
        </View>
      );
    } else {
      return null;
    }
  }
}

const position = {
  y: PropTypes.number.isRequired
}

PopupMenu.propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.shape(position).isRequired,
}

export default PopupMenu;
