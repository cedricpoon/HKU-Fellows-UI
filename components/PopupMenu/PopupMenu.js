import React, { Component } from 'react';
import PropTypes from "prop-types";
import { TouchableWithoutFeedback, View, Dimensions } from 'react-native';
import * as Animatable from 'react-native-animatable';

import { mapLayoutToState } from 'hkufui/components/helper';
import styles from './Styles';
import { footer } from 'hkufui/theme/grid';

class PopupMenu extends Component {
  constructor(props) {
    super(props);

    /* init width for first time rendering */
    const _menuLayout = {};
    _menuLayout['width'] = 0;
    this.state = { menuLayout: _menuLayout, toggleFlag: false };
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({ toggleFlag: !this.state.toggleFlag });
  }

  render() {
    const { children, position, parentHeight } = this.props;
    const { menuLayout, toggleFlag } = this.state;
    const { width, height } = Dimensions.get('window');

    if (toggleFlag) {
      const conceal = () => {
        this._menu.zoomOut(100).then(this.toggle);
      }

      /* Calculate position */
      const appendPos = { left: position.x, top: parentHeight + footer.popUpMargin };
      if (position.x <= 0) {
        appendPos.left = footer.popUpMargin;
      } else if (position.x + menuLayout.width >= width) {
        appendPos.left = width - menuLayout.width - footer.popUpMargin;
      }

      if (position.y < 0) {
        appendPos.top = appendPos.top + menuLayout.height;
      } else if (position.y + menuLayout.height >= height) {
        appendPos.top = -menuLayout.height - footer.popUpMargin
      }

      return(
        <View style={ styles.container }>
          <TouchableWithoutFeedback onPress={conceal}>
            <View
              style={[
                styles.outsider,
                {top: -position.y}
              ]}
            ></View>
          </TouchableWithoutFeedback>

          <Animatable.View
            style={[
              styles.menu,
              { ...appendPos }
            ]}
            animation="zoomIn"
            duration={100}
            onLayout={mapLayoutToState("menuLayout", this)}
            ref={ref => { this._menu = ref }}
          >
            { children }
          </Animatable.View>
        </View>
      );
    } else {
      return null;
    }
  }
}

const position = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired
}

PopupMenu.propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.shape(position).isRequired,
  parentHeight: PropTypes.number
}

export default PopupMenu;
