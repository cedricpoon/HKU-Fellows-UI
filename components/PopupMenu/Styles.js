import { StyleSheet, Dimensions } from 'react-native';

import { logo } from 'hkufui/theme/palette';
import { footer } from 'hkufui/theme/grid';

export default StyleSheet.create({
  container: {
    /* z-index in <PreviewFooter /> */
    zIndex: 2
  },
  menu: {
    /* z-index in <PopupMenu /> */
    zIndex: 2,
    position: 'absolute',
    backgroundColor: 'white',
    borderRadius: footer.borderRadius,
    borderColor: 'lightgrey',
    borderWidth: 1,
    shadowColor: logo.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: footer.shadowRadius
  },
  menuAndroid: {
    zIndex: 2,
    backgroundColor: 'white',
    borderColor: 'lightgrey',
    borderWidth: 1,
  },
  outsider: {
    /* z-index in <PopupMenu /> */
    zIndex: 1,
    position: 'absolute',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height
  }
});
