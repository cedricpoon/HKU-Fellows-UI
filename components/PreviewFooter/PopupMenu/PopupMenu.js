import React, { Component } from 'react';
import PropTypes from "prop-types";
import { TouchableWithoutFeedback, View } from 'react-native';
import * as Animatable from 'react-native-animatable';

import styles from './Styles';

class PopupMenu extends Component {

  constructor(props) {
    super(props);

    this.state = { width: 0 }
    this.setWidthToState = this.setWidthToState.bind(this);
  }

  setWidthToState(event) {
    this.setState({
      width: event.nativeEvent.layout.width
    });
  }

  render() {
    const { children, position, parentHeight, toggle, rightSided } = this.props;
    const { width } = this.state;

    const conceal = () => {
      this.menu.zoomOut(100).then(toggle);
    }

    return(
      <View style={ styles.container }>
        <TouchableWithoutFeedback onPress={conceal}>
          <View
            style={[
              styles.outsider,
              { top: styles.outsider.top + parentHeight }
            ]}
          ></View>
        </TouchableWithoutFeedback>

        <Animatable.View
          style={[
            styles.menu,
            {
              left: position - (rightSided ? width : 0),
              bottom: parentHeight + 10
            }
          ]}
          animation="zoomIn"
          duration={100}
          onLayout={this.setWidthToState}
          ref={ref => { this.menu = ref }}
        >
          { children }
        </Animatable.View>
      </View>
    );
  }
}

PopupMenu.propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.number.isRequired,
  parentHeight: PropTypes.number,
  rightSided: PropTypes.bool,
  toggle: PropTypes.func.isRequired
}

export default PopupMenu;
