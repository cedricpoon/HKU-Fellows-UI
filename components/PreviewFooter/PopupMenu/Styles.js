import { StyleSheet, Dimensions } from 'react-native';

import color from 'hkufui/theme/palette';

export default StyleSheet.create({
  container: {
    /* z-index in <PreviewFooter /> */
    zIndex: 2
  },
  menu: {
    /* z-index in <PopupMenu /> */
    zIndex: 2,
    position: 'absolute',
    backgroundColor: 'white',
    borderRadius: 10,
    borderColor: 'lightgrey',
    borderWidth: 1,
    shadowColor: color.logo.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 5
  },
  outsider: {
    /* z-index in <PopupMenu /> */
    zIndex: 1,
    position: 'absolute',
    top: -Dimensions.get('window').height,
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height
  }
});
