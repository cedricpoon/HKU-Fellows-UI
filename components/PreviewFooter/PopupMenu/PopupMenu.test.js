import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import PopupMenu from './PopupMenu';

describe('Testing PopupMenu Component', () => {
  it('renders as expected with required props', () => {
    const wrapper = shallow(
      <PopupMenu position={0} toggle={() => {}}>
        <View></View>
      </PopupMenu>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with required props and parentHeight', () => {
    const wrapper = shallow(
      <PopupMenu position={0} toggle={() => {}} parentHeight={0}>
        <View></View>
      </PopupMenu>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
