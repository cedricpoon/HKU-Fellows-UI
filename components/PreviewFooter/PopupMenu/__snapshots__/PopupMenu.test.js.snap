// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PopupMenu Component renders as expected with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopupMenu
    position={0}
    toggle={[Function]}
  >
    <Component />
  </PopupMenu>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Component
            style={
              Array [
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "top": -1334,
                  "width": 750,
                  "zIndex": 1,
                },
                Object {
                  "top": NaN,
                },
              ]
            }
          />
        </TouchableWithoutFeedback>,
        <withAnimatable(Component)
          animation="zoomIn"
          delay={0}
          direction="normal"
          duration={100}
          iterationCount={1}
          iterationDelay={0}
          onAnimationBegin={[Function]}
          onAnimationEnd={[Function]}
          onLayout={[Function]}
          onTransitionBegin={[Function]}
          onTransitionEnd={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "lightgrey",
                "borderRadius": 10,
                "borderWidth": 1,
                "position": "absolute",
                "shadowColor": "#4E4A47",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 5,
                "zIndex": 2,
              },
              Object {
                "bottom": NaN,
                "left": 10,
              },
            ]
          }
          useNativeDriver={false}
        >
          <Component />
        </withAnimatable(Component)>,
      ],
      "style": Object {
        "zIndex": 2,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component
            style={
              Array [
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "top": -1334,
                  "width": 750,
                  "zIndex": 1,
                },
                Object {
                  "top": NaN,
                },
              ]
            }
          />,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Array [
              Object {
                "height": 1334,
                "position": "absolute",
                "top": -1334,
                "width": 750,
                "zIndex": 1,
              },
              Object {
                "top": NaN,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": "zoomIn",
          "children": <Component />,
          "delay": 0,
          "direction": "normal",
          "duration": 100,
          "easing": undefined,
          "iterationCount": 1,
          "iterationDelay": 0,
          "onAnimationBegin": [Function],
          "onAnimationEnd": [Function],
          "onLayout": [Function],
          "onTransitionBegin": [Function],
          "onTransitionEnd": [Function],
          "style": Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "lightgrey",
              "borderRadius": 10,
              "borderWidth": 1,
              "position": "absolute",
              "shadowColor": "#4E4A47",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "zIndex": 2,
            },
            Object {
              "bottom": NaN,
              "left": 10,
            },
          ],
          "transition": undefined,
          "useNativeDriver": false,
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Component
              style={
                Array [
                  Object {
                    "height": 1334,
                    "position": "absolute",
                    "top": -1334,
                    "width": 750,
                    "zIndex": 1,
                  },
                  Object {
                    "top": NaN,
                  },
                ]
              }
            />
          </TouchableWithoutFeedback>,
          <withAnimatable(Component)
            animation="zoomIn"
            delay={0}
            direction="normal"
            duration={100}
            iterationCount={1}
            iterationDelay={0}
            onAnimationBegin={[Function]}
            onAnimationEnd={[Function]}
            onLayout={[Function]}
            onTransitionBegin={[Function]}
            onTransitionEnd={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "lightgrey",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "position": "absolute",
                  "shadowColor": "#4E4A47",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 5,
                  "zIndex": 2,
                },
                Object {
                  "bottom": NaN,
                  "left": 10,
                },
              ]
            }
            useNativeDriver={false}
          >
            <Component />
          </withAnimatable(Component)>,
        ],
        "style": Object {
          "zIndex": 2,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component
              style={
                Array [
                  Object {
                    "height": 1334,
                    "position": "absolute",
                    "top": -1334,
                    "width": 750,
                    "zIndex": 1,
                  },
                  Object {
                    "top": NaN,
                  },
                ]
              }
            />,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Array [
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "top": -1334,
                  "width": 750,
                  "zIndex": 1,
                },
                Object {
                  "top": NaN,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": "zoomIn",
            "children": <Component />,
            "delay": 0,
            "direction": "normal",
            "duration": 100,
            "easing": undefined,
            "iterationCount": 1,
            "iterationDelay": 0,
            "onAnimationBegin": [Function],
            "onAnimationEnd": [Function],
            "onLayout": [Function],
            "onTransitionBegin": [Function],
            "onTransitionEnd": [Function],
            "style": Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "lightgrey",
                "borderRadius": 10,
                "borderWidth": 1,
                "position": "absolute",
                "shadowColor": "#4E4A47",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 5,
                "zIndex": 2,
              },
              Object {
                "bottom": NaN,
                "left": 10,
              },
            ],
            "transition": undefined,
            "useNativeDriver": false,
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing PopupMenu Component renders as expected with required props and parentHeight 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopupMenu
    parentHeight={0}
    position={0}
    toggle={[Function]}
  >
    <Component />
  </PopupMenu>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Component
            style={
              Array [
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "top": -1334,
                  "width": 750,
                  "zIndex": 1,
                },
                Object {
                  "top": -1334,
                },
              ]
            }
          />
        </TouchableWithoutFeedback>,
        <withAnimatable(Component)
          animation="zoomIn"
          delay={0}
          direction="normal"
          duration={100}
          iterationCount={1}
          iterationDelay={0}
          onAnimationBegin={[Function]}
          onAnimationEnd={[Function]}
          onLayout={[Function]}
          onTransitionBegin={[Function]}
          onTransitionEnd={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "lightgrey",
                "borderRadius": 10,
                "borderWidth": 1,
                "position": "absolute",
                "shadowColor": "#4E4A47",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 5,
                "zIndex": 2,
              },
              Object {
                "bottom": 10,
                "left": 10,
              },
            ]
          }
          useNativeDriver={false}
        >
          <Component />
        </withAnimatable(Component)>,
      ],
      "style": Object {
        "zIndex": 2,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component
            style={
              Array [
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "top": -1334,
                  "width": 750,
                  "zIndex": 1,
                },
                Object {
                  "top": -1334,
                },
              ]
            }
          />,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Array [
              Object {
                "height": 1334,
                "position": "absolute",
                "top": -1334,
                "width": 750,
                "zIndex": 1,
              },
              Object {
                "top": -1334,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": "zoomIn",
          "children": <Component />,
          "delay": 0,
          "direction": "normal",
          "duration": 100,
          "easing": undefined,
          "iterationCount": 1,
          "iterationDelay": 0,
          "onAnimationBegin": [Function],
          "onAnimationEnd": [Function],
          "onLayout": [Function],
          "onTransitionBegin": [Function],
          "onTransitionEnd": [Function],
          "style": Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "lightgrey",
              "borderRadius": 10,
              "borderWidth": 1,
              "position": "absolute",
              "shadowColor": "#4E4A47",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "zIndex": 2,
            },
            Object {
              "bottom": 10,
              "left": 10,
            },
          ],
          "transition": undefined,
          "useNativeDriver": false,
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <Component
              style={
                Array [
                  Object {
                    "height": 1334,
                    "position": "absolute",
                    "top": -1334,
                    "width": 750,
                    "zIndex": 1,
                  },
                  Object {
                    "top": -1334,
                  },
                ]
              }
            />
          </TouchableWithoutFeedback>,
          <withAnimatable(Component)
            animation="zoomIn"
            delay={0}
            direction="normal"
            duration={100}
            iterationCount={1}
            iterationDelay={0}
            onAnimationBegin={[Function]}
            onAnimationEnd={[Function]}
            onLayout={[Function]}
            onTransitionBegin={[Function]}
            onTransitionEnd={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "lightgrey",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "position": "absolute",
                  "shadowColor": "#4E4A47",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 5,
                  "zIndex": 2,
                },
                Object {
                  "bottom": 10,
                  "left": 10,
                },
              ]
            }
            useNativeDriver={false}
          >
            <Component />
          </withAnimatable(Component)>,
        ],
        "style": Object {
          "zIndex": 2,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component
              style={
                Array [
                  Object {
                    "height": 1334,
                    "position": "absolute",
                    "top": -1334,
                    "width": 750,
                    "zIndex": 1,
                  },
                  Object {
                    "top": -1334,
                  },
                ]
              }
            />,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Array [
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "top": -1334,
                  "width": 750,
                  "zIndex": 1,
                },
                Object {
                  "top": -1334,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": "zoomIn",
            "children": <Component />,
            "delay": 0,
            "direction": "normal",
            "duration": 100,
            "easing": undefined,
            "iterationCount": 1,
            "iterationDelay": 0,
            "onAnimationBegin": [Function],
            "onAnimationEnd": [Function],
            "onLayout": [Function],
            "onTransitionBegin": [Function],
            "onTransitionEnd": [Function],
            "style": Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "lightgrey",
                "borderRadius": 10,
                "borderWidth": 1,
                "position": "absolute",
                "shadowColor": "#4E4A47",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 5,
                "zIndex": 2,
              },
              Object {
                "bottom": 10,
                "left": 10,
              },
            ],
            "transition": undefined,
            "useNativeDriver": false,
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
